rust/operator-binary/src/listener_controller.rs

```diff
@@ -172,17 +184,9 @@ pub async fn reconcile(listener: Arc<Listener>, ctx: Arc<Ctx>) -> Result<control
             ..Default::default()
         },
         spec: Some(ServiceSpec {
-            type_: Some(match listener_class.spec.service_type {
-                ServiceType::NodePort => "NodePort".to_string(),
-                ServiceType::LoadBalancer => "LoadBalancer".to_string(),
-                ServiceType::ClusterIP => "ClusterIP".to_string(),
-            }),
+            type_: Some(listener_class.spec.service_type.to_string()),
```

Comment:
Oh.. right.. I think I remember why this was the way it was now.

Listeners _currently happens_ to support the same set of service types as upstream k8s, but we still want to have the freedom to add custom ones in the future (for example: Istio ingress?).

Reply:
WDYT of putting your comment as comment below the line (I didn't know about this) and leave it as is until we need to change?

Reply:
IMO: No, I'd want the build to fail when a new unhandled variant has been introduced. It's also an argument for changing the name back from `KubernetesServiceType` (should've caught this back during the op-rs review.. sorry!).

Reply:
ptal

Reply:
ptal?

Reply:
Please take another look :)
This PR now points against a version of operator-rs where we renamed the struct back